packages/components/src/components/hds/app-side-nav/list/title.hbs

```diff
@@ -0,0 +1,11 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<Hds::AppSideNav::List::Item>
+  <div
```

Comment:
This is valid but let's discuss more. @hashicorp/hds-engineering 

We could possibly do this as a follow on as well.

Reply:
The hint in the class name (`hds-app-side-nav__list-title`) would suggest the same. We can write a follow-up task for this if you prefer, Kristin

Reply:
I would prefer to do this as a follow on as this is how it was in the SideNav. (I want to be able to wrap up the main work related to visual details for this PR and then do improvements like this as a fast follow on task.)

Reply:
I'm making a list and will add this to it as a follow on.

Reply:
I agree this needs to be done but I would like to make these kind of updates as a follow on since what it has now is what is currently already implemented in the SideNav.

Doing it as a follow on would make it easier for others to see what's been changed and to review since this current PR takes the majority of its code directly from the current SideNav code (which is a lot of code).

Reply:
Started a new branch for "follow-ons". Please take a look and let me know any feedback

- Branch: https://github.com/hashicorp/design-system/pull/2463
- Commit: https://github.com/hashicorp/design-system/pull/2463/commits/dbe1568f4afebae23ac02e55ad80ee4f6913d6a8

(I'll do the smaller updates like the above just in this one branch and if there's anything more involved I may create other branches...)